lib/rollbar/plugin.rb

```diff
@@ -15,31 +17,73 @@ def initialize(name)
       @name = name
       @dependencies = []
       @callables = []
+      @revert_callables = []
       @loaded = false
+      @on_demand = false
+    end
+
+    def load_on_demand
+      @on_demand = true
     end
 
     def configuration
       Rollbar.configuration
     end
 
+    def load_scoped!(transparent = false)
+      if transparent
+        load! if load?
+
+        result = yield
+
+        unload! if loaded
+      else
+        return unless load?
+
+        load!
+
+        result = yield
+
+        unload!
+      end
+
+      result
+    end
+
     def load!
       return unless load?
 
       begin
         callables.each(&:call)
-      rescue => e
+      rescue StandardError => e
         log_loading_error(e)
       ensure
         self.loaded = true
       end
     end
 
+    def unload!
+      return unless loaded
+
+      begin
+        revert_callables.each(&:call)
+      rescue StandardError => e
+        log_loading_error(e)
```

Comment:
It would be nice if this had a different message when failing on unload. e.g. "Error trying to unload plugin"

Reply:
👍